### PR TITLE
[IPU] Remove convert_to_half, suggest using `model.half`

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

As discussed in #7923 use `model.half` for pure FP16 training rather than having a specific argument for the plugin.

Fixes #<issue_number>

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: tests/accelerators/test_ipu.py [hunk 1]
```diff
@@ -200,20 +200,13 @@ class TestCallback(Callback):
...
     model = IPUModel()
-    trainer = Trainer(
-        default_root_dir=tmpdir,
-        fast_dev_run=True,
-        ipus=1,
-        precision=16,
-        plugins=IPUPlugin(convert_model_to_half=True),
-        callbacks=TestCallback()
-    )
+    model = model.half()
```
Comment (kaushikb11) [inline tests/accelerators/test_ipu.py:208]: Curious. Why not half it as a flag?

Comment (SeanNaren) [reply tests/accelerators/test_ipu.py:208]: its a bit boilerplate-ish and not very pytorch-esque. It's a bit confusing why you can't just convert your model to half which is the pytorch way and what we really do internally :)

**Review by kaushikb11 — APPROVED** [review]
